lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -2604,6 +2604,96 @@ class ConvertAtenPoolingOp : public ConvertAtenPoolingBaseOp<AtenOpT, TosaOpT> {
   }
 };
 
+// Ref: Error checking based on the Torch to LinAlg lowering
+template <typename AtenOpT, int fillVal>
+class ConvertAtenConstPatternOp : public OpConversionPattern<AtenOpT> {
+public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+  LogicalResult
+  matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    auto outType =
+        static_cast<Type>(
```

Comment:
Removal of that static_cast gives a type inference error with gcc-7.5.0:

..................error: expected primary-expression before '>' token
                        .dyn_cast<TensorType>();

The other possibility is to collect the output of 'convertType' in a variable and then gcc-7.5.0 is happy. I thought that the static_cast is simpler and avoids the unnecessary creation of a variable.

Reply:
do you mean changing the code to 
```
OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
                op.getType())).template dyn_cast<TensorType>();
```
would cause the error?

Reply:
Good catch! I had missed the .template and then never went back to checking on the error. 
Pushed a new one with the fixes.